### ISSUE TITLE
Update new-hub.rst and cookiecutter template

### ISSUE THREAD

**Issue by ryanlovett** [berkeley-dsep-infra/datahub#2062]
I deployed stat159-staging today by following the new-hub docs and came across some items there and in the cookiecutter template that I think should be updated. I'll patch everything if there's no other comments:

1. The cookiecutter template suggests "dns-valid-hub-name" as a value for "hub_name". It really means the short hostname (the name before the first dot), and not the fqdn or "dns-valid". I recommend that the hint be changed to "dns-hub-short-name".
2. The template needs to change nfsPVC.nfs.serverIP to `nfsserver-01`.
3. Between steps 4 and 5, you have to 1) add the new deployment to `.circleci/config.yml`, and 2) create new k8s namespaces manually. This used to be done by helm2.
4. `secrets/gcr-key.json` and `secrets/gke-key.json` are not part of the cookiecutter, so they need to be copied over from one of the other deployments.
5. Step 7 should refer to config/staging.yaml, not secrets/staging.yaml
6. The template needs a new jupyterhub-ssh.hostKey variable in each secrets file with `ssh-keygen` hint, along with jupyterhub-ssh.hubUrl in the staging/prod config files.
7. Step 9 needs to be done before step 5 due to jupyterhub-ssh
8. Add doc about canvas oauth client id and secrets. From @felder, "so datahub has two developer keys in canvas. one is for all of the staging hubs (they all share a key). one is for the prod hubs (they all share a key)". we should ask @felder to ask Miles to update the hub list in bcourses. If @felder is not available, we should email bcourseshelp@berkeley.edu with a specific mention of Miles or a cc to him.
9. Document adding new netreg offsite hosting entries for prod/staging in the "CDSS, RTL Supported" security context.

**Comment by felder** [comment]
@ryanlovett Miles is ok with me updating the list myself.  So generally speaking, just ask me and I'll make the change.  In the event I'm not available, then send the email to bcourseshelp@berkeley.edu

**Comment by felder** [comment]
I too found the cookiecutter hint "dns-valid-hub-name" to be confusing.  I think adding something like "dns-valid-short-name (e.g. datahub, r, or data100)" would be helpful here.

**Comment by felder** [comment]
cookiecutter template needs to set images.images.path and images.images.name in hubploy.yaml

**Comment by felder** [comment]
jupyterhub.auth.type needs to be changed from google to custom to use bcourses auth

**Comment by yuvipanda** [comment]
This has been updated now!
